docs/language/source/specification_language_description.rst

```diff
@@ -850,7 +850,7 @@ The specification of a datasets is described in YAML as follows:
         neurodata_type_inc: Optional neurodata_type the group should inherit from
         quantity: Optional quantity identifier for the group (default=1).
         linkable: Boolean indicating whether the group is linkable (default=True)
-        dtype: Required string describing the data type of the dataset
+        dtype: Optional string describing the data type of the dataset
```

Comment:
We should add "(default=null)" to clarify that when dtype is not specified that this is the same as ``dtype: null``.